document/JS.bs

```diff
@@ -0,0 +1,935 @@
+</div>
+
+<div algorithm>
+  The <dfn method for="WebAssembly">instantiate(moduleObject, importObject)</dfn> method, when invoked, performs the following steps:
+    1. Let |promise| be [=a new promise=].
+    1. [=Queue a task=] to perform the following steps:
+        1. [=instantiate a WebAssembly module|Instantiate the WebAssembly module=] |module| importing |importObject|, and let |instance| be the result.  If this throws an exception, catch it, and [=reject=] |promise| with the exception.
+        1. [=Resolve=] |promise| with |instance|.
+           1. Other unspecified asynchronous tasks may run at this point.
+    1. Return |promise|
+</div>
+
+
+Note: A follow-on streaming API is documented [in the WebAssembly design repository](https://github.com/WebAssembly/design/blob/master/Web.md#additional-web-embedding-api).
+
+<h3 id="modules">Modules</h3>
+
+<pre class="idl">
+enum ImportExportKind {
+  "function",
+  "table",
+  "memory",
+  "global"
+};
+
+dictionary ModuleExportDescriptor {
+  required DOMString name;
+  required ImportExportKind kind;
+  // Note: Other fields such as signature may be added in the future.
+};
+
+dictionary ModuleImportDescriptor {
+  required DOMString module;
+  required DOMString name;
+  required ImportExportKind kind;
+};
+
+[LegacyNamespace=WebAssembly, Contructor(BufferSource bytes), Exposed=(Window, Worker)]
+interface Module {
+  static sequence&lt;ModuleExportDescriptor> exports(Module module);
+  static sequence&lt;ModuleImportDescriptor> imports(Module module);
+  static sequence&lt;ArrayBuffer> customSections(Module module, DOMString sectionName);
+};
+</pre>
+
+<!-- Any use for this?
+<div algorithm="name of the export">
+    The <dfn>name of the export</dfn> |export| in the module |module| is defined as follows:
+    1. If |export| is of the form [=𝖿𝗎𝗇𝖼=] |funcidx| and |module| has a [=name section=] with a [=function names subsection=],
+         1. Within <code>[=funcnamesubsec=]</code>, search for an entry <code>[=nameassoc=]</code> of the form |funcidx| <code>name</code>. If such an entry is found, return <code>name</code> decoded as a UTF-8 into a JavaScript String.
+    1. Return the empty String.
+</div>
+-->
+
+<div algorithm>
+    The <dfn method for="Module">exports(module)</dfn> method, when invoked, performs the following steps:
+    1. Let |ast| be |module|.\[[Module]].
+    1. Let |exports| be an empty [=list=].
+    1. For each |export| in |ast|.exports:
+        1. Let |name| be |export|.name as a String.
+        1. Let |kind| be
+            * `"function"` if |export|.desc is of the form 𝖿𝗎𝗇𝖼 funcidx
+            * `"table"` if |export|.desc is of the form 𝗍𝖺𝖻𝗅𝖾 tableidx
+            * `"memory"` if |export|.desc is of the form 𝗆𝖾𝗆    memidx
+            * `"global"` if |export|.desc is of the form 𝗀𝗅𝗈𝖻𝖺𝗅 globalidx
+        1. Let |obj| be ! ObjectCreate(%ObjectPrototype%).
```

Comment:
Dictionary instances are not JS objects, so you should not be creating JS objects here. Instead you can return an Infra struct, or just something like "a ModuleExportDescriptor whose name is |name| and kind is |kind|".

Reply:
Used something like the second wording.

Reply:
Should I use ordered map operations on the import and export objects? Treating them this way probably will be an observable change, on the importing side, as it was based on getting specific things that were asked for. If I don't make that change, is there any IDL i can use to stand for an uninterpreted object?

Reply:
Can you explain the problem a bit more? How would it be an observable change? What change are you proposing, now that this seems to be fixed?